src/materials/nodes/manager/NodeMaterialObserver.js

```diff
@@ -224,6 +389,21 @@ class NodeMaterialObserver {
 
 		}
 
+		// geometry
+
+		const geometryData = renderObjectData.geometry;
+		const currentGeometryData = this.getGeometryData( renderObject.geometry );
```

Comment:
Could we compare geometry data without get a new data every frame?

Reply:
Good point, done!